src/test/java/io/jenkins/blueocean/autofavorite/FavoritingScmListenerTest.java

```diff
@@ -95,12 +109,18 @@ private WorkflowJob createAndRunPipeline() throws java.io.IOException, Interrupt
 
         WorkflowJob job = mbp.getItem("master");
         while (job.getBuilds().isEmpty()) {
-            Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+            Thread.sleep(5);
         }
 
         WorkflowRun run = job.getBuildByNumber(1);
         assertNotNull(run);
 
+        if (removeBuildData) {
+            while (!run.removeActions(BuildData.class)) {
```

Comment:
Potential for infinite loop? Maybe add a timeout or max retries? 🐜

Reply:
Yea, I'll figure out a way to make sure that doesn't happen. removeActions() returns true once it modifies the list, so we want to keep trying until we get it, but if the action never gets added then this will (obviously) never give up.

Reply:
I updated the test case to combine both checks for the build data removal and for waiting until the test finishes.